### PR TITLE
Private MessageChannels: split capability and add dataLayer.getDataLayerPort

### PR DESCRIPTION
**Pull request by kerrynf**
## Description

This PR will rework the private MessageChannels API to properly define sub capabilities and add in a second sub capability for dataLayer, adding getDataLayerPort.

### Main changes in the PR:

1. Rework MessageChannels capability to support separate sub capabilities per channel type
2. Add a second channel type for dataLayer

### Validation performed:

1. Updated the test apps
2. Added and updated the unit tests

### Unit Tests added:

Yes

### End-to-end tests added:

No - E2E will need to be added as fast follow to manage versioning challenges around the refactor.

### Change file added:

Yes

### Related PRs:

[#2140](https://github.com/OfficeDev/microsoft-teams-library-js/pull/2140)

### Next/remaining steps:

- E2E tests will be added as fast follow once the version updates are available.

### CODE REVIEW

File: packages/teams-js/test/private/messageChannels.spec.ts
Comment (sumathin-lang) [inline packages/teams-js/test/private/messageChannels.spec.ts:72]: whats the value in running through each frameContext? the code under test doesnt use this

File: change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json
Comment (sumathin-lang) [inline change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json:6]: i assume it is okay to mark as patch even though it is a breaking change since this is internal API?

**Review by sumathin-lang — APPROVED** [review]

File: change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json
Comment (kerrynf) [reply change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json:6]: It isn't a patch, it is minor bump (the top type is the type to look at here). Trevor said that since these APIs are marked beta the breaking changes are not an issue (this would be true even if they weren't private I assume).

`"type": "minor",`

File: packages/teams-js/test/private/messageChannels.spec.ts
Comment (kerrynf) [reply packages/teams-js/test/private/messageChannels.spec.ts:72]: This was a requirement from the teams-js team on my previous PR (I had the same question). Some APIs change based on frame context, so it was recommended to make sure we are testing this just in case some one starts introducing changes in this area.

Comment (TrevorJoelHarris) [reply packages/teams-js/test/private/messageChannels.spec.ts:72]: Yes, kerryn is correct. Really, sumathi is too and for this particular code right now there's no difference. Testing for each framecontext will make sure that if someone goes and removes one in the future it will break a test (which is good because removing a frame context would be a breaking change and we don't want to do that accidentally).

Ultimately, I would like to get rid of the framecontext concept entirely and depend entirely on isSupported, allowing the hosts to control what is and isn't supported based on their knowledge of where the app is running, but there are one or two things we'd need to change before we get there

File: change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json
Comment (TrevorJoelHarris) [reply change/@microsoft-teams-js-f6daee5c-d395-44ed-a4f3-670c3bbc44fe.json:6]: Correct, if something is marked as beta we reserve the right to change the signature with little or no notice.

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

**Comment by TrevorJoelHarris** [comment]
This looks good to me, will approve once you have a signed off security review

**Review by TrevorJoelHarris — APPROVED** [review]

File: packages/teams-js/src/internal/telemetry.ts
Comment (shrshindeMSFT) [inline packages/teams-js/src/internal/telemetry.ts:202]: Can we alphabetize this?

File: packages/teams-js/src/internal/telemetry.ts
Comment (shrshindeMSFT) [inline packages/teams-js/src/internal/telemetry.ts:201]: Also, can you explain how will the telemetry will be handled from previous `messageChannels.getTelemetryPort` to `messageChannels.telemetry.getTelemetryPort`? Can't we use the existing one?

**Review by jadahiya-MSFT — APPROVED** [review]

File: packages/teams-js/src/private/messageChannels.ts
Comment (shrshindeMSFT) [inline packages/teams-js/src/private/messageChannels.ts:17]: Why do we need nested namespace?

**Pull request merged**
